main/2021/CG-08-03.md

```diff
@@ -43,4 +43,340 @@ Installation is required, see the calendar invite.
+DS: AR, WO are the “against” votes:  anything you want to note, especially that is not already a part of the presentation?
+
+AR: the desire for UTF-16 lowering based on various assumptions about what implementations do, which are probably not what modern implementations do. Think that we don’t have data to decide if it is a valuable optimization.
+
+DD (chat): Fair point, that downgrades me from F to N. (Not that matters much I imagine?)
+
+AvK (chat): (I was also wondering in what way the outcome influences things.)
+
+
+POLL: Poll for maintaining single list-of-USV `string` type
+
+
+SF: 31
+F: 8
+N: 6
+A: 0
+SA: 2
+
+DS: dcode, Joe, anything you would like to say?
+
+JP: This could be a hit for JS, maybe we could provide the fastest path possible for JS? Possibly valuable to have more discussions.
+
+DW (chat): This is really bad for AssemblyScript, for the reasons I have presented. Killed by specification. Congrats.
+
+CW (chat): Disagree, especially considering AS accidentally implemented sanitisation already.
+
+DW (chat): Nonsense!
+
+DD (chat) I guess AssemblyScript was dead from September 2020 through June 2021…
+
+DW (chat): Just add a domstring, doesn't harm what's anticipated :(
+
+JP (chat): Please don't use that argument Conrad: if someone already accidentally crashed without a seatbelt, should cars not have seat belts? Hence it is not a great argument.
+
+DW (chat): (lol @Domenic)
+
+DW (chat): Logically invalid. Fallacy.
+AT (chat): I think we can encourage a positive tone @dcode @Joe :)
+
+DW (chat): I think positive tone is over.
+
+ZB (chat): For langs like Java, AS & co it would be nice to have built-in string or other ways to have fast/cheap interop with browser API, but probably it’s something separate.
+
+AT (chat): We can figure thigns out back in the AS side of things :)
+
+DW (chat): This is not what the spec is supposed to do.
+
+DW (chat): I could not disagree more with this.
+
+JP (chat): I think both sides are being disrespectful, regardless of the outcome.
+
+DW (chat): You should account for concerns by implementers, not this.
+
+JP (chat): If language match with each other (f.e. JS + AS) why not have the best interface?
+Java + C#, Java + JS, etc.
+DD (chat): They can use one-off interfaces, but the purpose of standard interop is... standard interop.
+JP (chat): Will we want to make separate specs for new interfaces? Seems like more work, and not guaranteed
+
+
+KM: given the strings have to be copied, the transformation makes more sense. You can choose your behavior as a consumer. You can come back in the future and say, and hope doesn’t preclude data coming back that says some applications have problems under the sanitization world, and reconsider. But at this time it seems reasonable to start here.
+
+TT: I’ve tried to find a middle ground between both sides, I wonder if it’s possible within the components to have additional interface types that are not language agnostic. And can cater to specialized cases[?]
```

Comment:
I've tried to find a middle ground between both sides. I wonder if it’s possible that inside of a component to have additional interface types that are not language agnostic but can take into account shared encodings between modules.

Reply:
Oops, it was particularly your closing comment that prompted me to write https://github.com/WebAssembly/interface-types/issues/135#issuecomment-892082643 on this topic, but I forgot to tag you. Although not discussed in that post (because I ended up responding to a different comment), the technology should also be able to do the sort of bridging I believe you are suggesting. Sorry for forgetting to tag you and respond more directly!

Reply:
Thanks for the reference! That is exactly along the lines I was thinking as well. I would be more than interested in your thoughts about this and I think an intra component shared encoding linking model with potentially zero copy would be a great addition and fit into and complement the component model perfectly and make everyone a first class. I have a few rough starting points for this, but @littledan thoughts have some great overlap also with regards to externref potentially.